src/test/kotlin/uk/gov/dluhc/printapi/messaging/SendApplicationToPrintMessageListenerIntegrationTest.kt

```diff
@@ -189,4 +136,116 @@ internal class SendApplicationToPrintMessageListenerIntegrationTest : Integratio
+            supportingInformationFormat = supportingInfoFormat,
+            photoLocationArn = photoLocation,
+            delivery = with(delivery) {
+                Delivery(
+                    addressee = addressee,
+                    address = with(address) {
+                        Address(
+                            street = street,
+                            postcode = postcode,
+                            property = property,
+                            locality = locality,
+                            town = town,
+                            area = area,
+                            uprn = uprn
+                        )
+                    },
+                    deliveryClass = DeliveryClass.STANDARD,
+                    deliveryAddressType = DeliveryAddressType.REGISTERED,
+                    addressFormat = AddressFormat.UK,
+                )
+            },
+            eroEnglish = with(localAuthority) {
+                ElectoralRegistrationOffice(
+                    name = "Electoral Registration Officer",
+                    phoneNumber = contactDetailsEnglish.phone,
+                    emailAddress = contactDetailsEnglish.email,
+                    website = contactDetailsEnglish.website,
+                    address = with(contactDetailsEnglish.address) {
+                        Address(
+                            street = street,
+                            postcode = postcode,
+                            property = property,
+                            locality = locality,
+                            town = town,
+                            area = area,
+                            uprn = uprn
+                        )
+                    }
+                )
+            },
+            eroWelsh = with(localAuthority) {
+                ElectoralRegistrationOffice(
+                    name = "Swyddog Cofrestru Etholiadol",
+                    phoneNumber = contactDetailsWelsh!!.phone,
+                    emailAddress = contactDetailsWelsh!!.email,
+                    website = contactDetailsWelsh!!.website,
+                    address = with(contactDetailsWelsh!!.address) {
+                        Address(
+                            street = street,
+                            postcode = postcode,
+                            property = property,
+                            locality = locality,
+                            town = town,
+                            area = area,
+                            uprn = uprn
+                        )
+                    }
+                )
+            },
+            statusHistory = mutableListOf(
+                PrintRequestStatus(
+                    status = Status.PENDING_ASSIGNMENT_TO_BATCH,
+                    eventDateTime = Instant.now()
+                )
+            ),
+            userId = userId
+        )
+        return printRequest
+    }
+
+    private fun assertSavedCertificate(
```

Comment:
Would be it good to extract this assert to its own class so that it can be used for future asserts? (We do something similar in vca)

Reply:
Agreed, I'm a big fan of the assertJ API and custom assertions, and I dont think we make enough use of them generally (across all the squads/projects)
I'd like to see it implemented as a custom assertion, and I'd also like to see us get rid of the complexity re `Instant` comparisons with the `BiPredicate` (I vaguely remember refactoring an assertion (not sure which project) that used BiPredicate etc for time comparisons down to little more than the standard assertJ isEqualsTo ??)

Reply:
You updated `CertificateRepositoryIntegrationTest` but that works rounding issues by controlling the assigned `Instant` using second precision in saved entity `Instant.now().truncatedTo(SECONDS)` and most other fields are known for the saved entity.  So following was possible :
```
private val IGNORED_FIELDS = arrayOf(".*dateCreated")

            assertThat(actual)
                .usingRecursiveComparison()
                .ignoringFieldsMatchingRegexes(*IGNORED_FIELDS)
                .isEqualTo(expected)
```                

In the integration test the Mappers assign real timestamps (not artificially truncated for comparison convenience) and includes randomly assigned data such as Ids and event dates.  So tricks used in that test class cannot be applied in this Integration test.

If we want to not use the AssertJ recursive comparison (which seemed to be the pattern used on the project) I'll introduce one - so many properties to compare though :-(